optimum/gptq/quantizer.py

```diff
@@ -0,0 +1,571 @@
+        if self.module_name_preceding_first_block is not None and not isinstance(
+            self.module_name_preceding_first_block, list
+        ):
+            raise ValueError("block_name_to_quantize must be a list of string")
+
+    def to_dict(self):
+        """
+        Return the args in dict format
+        """
+        return copy.deepcopy(self.__dict__)
+
+    @classmethod
+    def from_dict(cls, config_dict: Dict[str, Any]):
+        """
+        Instantiates a `GPTQQuantizer` using config_dict as kwargs
+
+        Args:
+            config_dict (`Dict[str,Any]`):
+                quantization config
+
+        Returns:
+            `GPTQQuantizer`:  The quantizer object instantiated from those parameters.
+        """
+        return cls(**config_dict)
+
+    def _replace_by_quant_layers(self, module: nn.Module, names: List[str], name: str = ""):
+        """
+        Replace linear layers in `module` by `QuantLinear`
+
+        Args:
+            module (`nn.Module`):
+                Module to quantize
+            names (`List[str]`):
+                List of names of the module to quantize
+            name (`str`, *optional*, defaults to `""`):
+                To keep track of the name of the current module
+        """
+        QuantLinear = dynamically_import_QuantLinear(
+            use_triton=False, desc_act=self.desc_act, group_size=self.group_size
+        )
+        if isinstance(module, QuantLinear):
+            return
+        for attr in dir(module):
+            layer = getattr(module, attr)
+            name1 = name + "." + attr if name != "" else attr
+            if name1 in names:
+                device = get_device(layer)
+                delattr(module, attr)
+                if isinstance(layer, nn.Linear):
+                    in_features = layer.in_features
+                    out_features = layer.out_features
+                elif isinstance(layer, nn.Conv2d):
+                    in_features = layer.in_channels
+                    out_features = layer.out_channels
+                elif isinstance(layer, Conv1D):
+                    in_features = layer.weight.shape[0]
+                    out_features = layer.weight.shape[1]
+                if not (self.desc_act) or self.group_size == -1:
+                    new_layer = QuantLinear(
+                        self.bits, self.group_size, in_features, out_features, True, use_cuda_fp16=self.use_cuda_fp16
+                    )
+                else:
+                    new_layer = QuantLinear(self.bits, self.group_size, in_features, out_features, True)
+                new_layer.device = device
+                setattr(module, attr, new_layer.to(device))
+        for name1, child in module.named_children():
+            self._replace_by_quant_layers(child, names, name + "." + name1 if name != "" else name1)
+
+    @torch.no_grad()
+    def quantize_model(
+        self,
+        model: nn.Module,
+        tokenizer: Any,
```

Comment:
Why Any?

Reply:
I didn't want to put `AutoTokenizer` to be as generic as possible. Any idea for a better solution ?

Reply:
No sounds good!